packages/adblocker-electron-preload/src/index.ts

```diff
@@ -68,11 +44,11 @@ if (window === window.top && window.location.href.startsWith('devtools://') ===
     window.addEventListener(
       'DOMContentLoaded',
       () => {
-        DOM_MONITOR = new DOMMonitor((update) => {
+        DOM_MONITOR = new DOMMonitor(async (update) => {
           if (update.type === 'features') {
-            getCosmeticsFiltersUpdate({
-              ...update,
-            });
+            const response = await injectCosmeticFilters();
+            if (response.error) throw new Error(`error injecting updated cosmetic filters: ${response.error}`)
+            if (!(ACTIVE = response.active)) unload();
```

Comment:
Do we want to do unloading here? I don't think we need to unload the dom monitor.

Reply:
`active` from `FiltersEngine.getCosmeticFilters` means that if we want cosmetic filters or not. This code will not be reached at all.

```ts
    if (this.config.loadCosmeticFilters === false) {
      return {
        active: false,
        extended: [],
        scripts: [],
        styles: '',
      };
    }
```